addons/project/tests/test_project_base.py

```diff
@@ -104,3 +104,30 @@ def test_delete_project_with_tasks(self):
 
         with self.assertRaises(UserError):
             self.project_pigs.unlink()
+
+    def test_auto_assign_stages_when_importing_tasks(self):
+        self.assertFalse(self.project_pigs.type_ids)
+        self.assertEqual(len(self.project_goats.type_ids), 2)
+        first_stage, second_stage = self.project_goats.type_ids
+        self.env['project.task'].create({
+            'name': 'First Task',
+            'user_id': self.user_projectuser.id,
+            'project_id': self.project_pigs.id,
+            'stage_id': first_stage.id
+        })
+        self.assertEqual(self.project_pigs.type_ids, first_stage)
+        self.env['project.task'].create([
+            {
+                'name': 'Second Task',
+                'user_id': self.user_projectuser.id,
+                'project_id': self.project_pigs.id,
+                'stage_id': second_stage.id
+            },
+            {
+                'name': 'Third Task',
+                'user_id': self.user_projectuser.id,
+                'project_id': self.project_pigs.id,
+                'stage_id': first_stage.id
+            },
+        ])
+        self.assertEqual(self.project_pigs.type_ids, self.project_goats.type_ids)
```

Comment:
```suggestion
        self.env['project.task']._load_records_create([
            {
                'user_id': self.user_projectuser.id,
                'project_id': self.project_pigs.id,
                'stage_id': first_stage.id
            } for stage in self.project_goats.type_ids
        ])
        self.assertEqual(self.project_pigs.type_ids, self.project_goats.type_ids)
```
Isn't this enough?
Btw, I don't remember how data is supposed to be contained in the `values` param of _load_records_create. So it would have to be checked.

Reply:
name is mandatory but okay i will give them the same one

Reply:
checked, it the same as create